### PR TITLE
Add git config options

### PR DESCRIPTION
**Pull request by RobLoach**
This adds the ability to set git config options. Do you know why it's failing?

### CODE REVIEW

**Comment by kevinawoo** [comment]
Sidenote, how was it to create a test?

Anything that needs improvement?

**Comment by RobLoach** [comment]
> Sidenote, how was it to create a test?

Pretty good, I'm glad there's some sort of common standard in place to create tests. I still don't know why it isn't passing though.

File: tasks/build_control.js [hunk 1]
```diff
@@ -160,6 +161,15 @@ module.exports = function (grunt) {
       }
     }
 
+
+    // Initialize the git config
+    function initConfig() {
+      for (var key in options.config) {
+        shelljs.exec('git config "' + key + '" ' + options.config[key]);
```
Comment (kevinawoo) [inline tasks/build_control.js]: If your name is `John Doe`, it only picks up `John`. Just need to add some qoutes to the option.config[key]

```
shelljs.exec('git config "' + key + '" ' + '"' + options.config[key] '"');
```

File: test/scenarios/git config/repo/gruntfile.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module.exports = function (grunt) {
+  // add custom tasks
+  // NOTE: cwd is `test/mock`
+  grunt.loadTasks('../../tasks');
```
Comment (kevinawoo) [inline test/scenarios/git config/repo/gruntfile.js]: just one level higher. (I'll try to make this a bit easier somehow)
```
  grunt.loadTasks('../../../tasks');

```

File: README.md [hunk 3]
```diff
@@ -135,6 +135,15 @@ Default: `true`
 
 Make sure that every commit on the built code branch matches a commit on the main project branch. If the main project's working directory has uncommitted changes, a commit task will throw an error.
 
+
+#### config
+Type: `Array`
```
Comment (kevinawoo) [inline README.md]: This is type Object.

**Comment by kevinawoo** [comment]
After doing those changes, it should be good and I'll merge it in.

**Comment by kevinawoo** [comment]
On second thought, I went ahead and did those changes, so you should be good.

**Comment by RobLoach** [comment]
@kevinawoo Nice. Thanks a lot, was that all that was missing?

**Comment by kevinawoo** [comment]
Yep, the quotes and file path was the only problems.
Here's the diff for those https://github.com/robwierzbowski/grunt-build-control/compare/19e0312b6ed7e909def54b7d0b71b23f88497b00...edfecda53a2ba0ae3d3c10f8766d1b54dc55dcf0 if you're interested.

**Pull request closed without merging**